File: applications/luci-statistics/luasrc/model/cbi/luci_statistics/email.lua
--[[

Luci configuration model for statistics - collectd email plugin configuration
(c) 2008 Freifunk Leipzig / Jo-Philipp Wich <xm@leipzig.freifunk.net>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

$Id$

]]--

m = Map("luci_statistics")

-- collectd_email config section
s = m:section( NamedSection, "collectd_email", "luci_statistics" )

-- collectd_email.enable
enable = s:option( Flag, "enable" )
enable.default = 0

-- collectd_email.socketfile (SocketFile)
socketfile = s:option( Value, "SocketFile" )
socketfile.default = "/var/run/collect-email.sock"
socketfile:depends( "enable", 1 )

-- collectd_email.socketgroup (SocketGroup)
socketgroup = s:option( Value, "SocketGroup" )
socketgroup.default  = "nobody"
socketgroup.rmempty  = true
socketgroup.optional = true
socketgroup:depends( "enable", 1 )

-- collectd_email.socketperms (SocketPerms)
socketperms = s:option( Value, "SocketPerms" )
socketperms.default  = "0770"
socketperms.rmempty  = true
socketperms.optional = true
socketperms:depends( "enable", 1 )

-- collectd_email.maxconns (MaxConns)
maxconns = s:option( Value, "MaxConns" )
maxconns.default   = 5
maxconns.isinteger = true
maxconns.rmempty   = true
maxconns.optional  = true
maxconns:depends( "enable", 1 )

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/email.js
'use strict';
'require baseclass';
'require form';
'require tools.widgets as widgets';

return baseclass.extend({
	title: _('E-Mail Plugin Configuration'),
	description: _('The email plugin creates a unix socket which can be used to transmit email-statistics to a running collectd daemon. This plugin is primarily intended to be used in conjunction with Mail::SpamAssasin::Plugin::Collectd but can be used in other ways as well.'),

	addFormOptions: function(s) {
		var o;

		o = s.option(form.Flag, 'enable', _('Enable this plugin'));

		o = s.option(form.Value, 'SocketFile', _('Socket file'));
		o.default = '/var/run/collectd/email.sock';
		o.depends('enable', '1');

		o = s.option(widgets.GroupSelect, 'SocketGroup', _('Socket group'));
		o.default = 'nogroup';
		o.optional = true;
		o.depends('enable', '1');

		o = s.option(form.Value, 'SocketPerms', _('Socket permissions'));
		o.default = '0770';
		o.optional = true;
		o.depends('enable', '1');
		o.validate = function(section_id, v) {
			if (v == '')
				return true;

			if (!v.match(/^[0-7]{1,4}$/))
				return _('Expecting permssions in octal notation');

			return true;
		};

		o = s.option(form.Value, 'MaxConns', _('Maximum allowed connections'));
		o.datatype = 'range(1,16384)';
		o.default = '5';
		o.optional = true;
		o.depends('enable', '1');
	},

	configSummary: function(section) {
		if (section.SocketFile)
			return _('Awaiting email input at %s').format(section.SocketFile);
	}
});
